test/RunSpec.hs

```diff
@@ -29,6 +29,14 @@ withCurrentDirectory workingDir action = do
 main :: IO ()
 main = hspec spec
 
+
+removeLoadedPackageEnvironment :: String -> String
+#if __GLASGOW_HASKELL__ < 810
```

Comment:
Would it hurt to apply this filter unconditionally?

Reply:
My primary intention is here is to document why we need this.  So if at some point in the future we drop support for GHC < 8.10 we can remove this code.